gix-object/src/commit/decode.rs

```diff
@@ -21,52 +22,43 @@ pub fn message<'a, E: ParserError<&'a [u8]> + AddContext<&'a [u8]>>(i: &'a [u8])
                 .map(|err: E| err.add_context(i, "newline + <message>")),
         );
     }
-    let (i, _) = tag(NL)
+    preceded(NL, rest.map(|s: &[u8]| s.as_bstr()))
         .context("a newline separates headers from the message")
-        .parse_next(i)?;
-    Ok((&[], i.as_bstr()))
+        .parse_next(i)
```

Comment:
Explanation: Moving to declarative parsing where possible not because of any high moral ground or technical benefit but just because it makes the winnow 0.5 upgrades less intrusive